client/state/jetpack-connect/actions.js

```diff
@@ -200,5 +205,53 @@ export default {
 				} );
 			} );
 		};
+	},
+
+	validateSSONonce( siteId, ssoNonce ) {
+		return ( dispatch ) => {
+			debug( 'Attempting to validate SSO for ' + siteId );
+			dispatch( {
+				type: JETPACK_CONNECT_SSO_VALIDATE,
+				siteId
+			} );
+
+			return wpcom.undocumented().jetpackValidateSSONonce( siteId, ssoNonce ).then( ( data ) => {
+				dispatch( {
+					type: JETPACK_CONNECT_SSO_VALIDATION_RECEIVE,
```

Comment:
This is optional, but a common pattern is to have 4 action types for a given request.
```
JETPACK_CONNECT_SSO_VALIDATION_REQUEST,
JETPACK_CONNECT_SSO_VALIDATION_RECEIVE,
JETPACK_CONNECT_SSO_VALIDATION_SUCCESS,
JETPACK_CONNECT_SSO_VALIDATION_ERROR
```

`JETPACK_CONNECT_SSO_VALIDATION_REQUEST` is dispatched like you have here immediately. While `JETPACK_CONNECT_SSO_VALIDATION_RECEIVE` and `JETPACK_CONNECT_SSO_VALIDATION_SUCCESS` are dispatched on success. Normally the receive action has the data payload, while we can use the `JETPACK_CONNECT_SSO_VALIDATION_SUCCESS` action to keep track of whether our action has completed. On error, we dispatch the `JETPACK_CONNECT_SSO_VALIDATION_ERROR` action with the error payload.

This may seem like an excessive number of action types, but it helps keep the reducer data simple. (See https://github.com/Automattic/wp-calypso/blob/master/client/state/sites/media-storage/reducer.js for an example) In this case, since we're only interested in a boolean value, I'd probably suggest using:
```
JETPACK_CONNECT_SSO_VALIDATION_REQUEST,
JETPACK_CONNECT_SSO_VALIDATION_SUCCESS,
JETPACK_CONNECT_SSO_VALIDATION_ERROR
```